presto-main/src/main/java/io/prestosql/sql/planner/LocalExecutionPlanner.java

```diff
@@ -2115,6 +2135,24 @@ private DynamicFilterSourceOperatorFactory createDynamicFilterSourceOperatorFact
                     });
         }
 
+        private Optional<LocalDynamicFilter> createDynamicFilter(SemiJoinNode node, LocalExecutionPlanContext context)
+        {
+            if (!isEnableDynamicFiltering(context.getSession()) || !node.getDynamicFilterId().isPresent()) {
```

Comment:
`isEnableDynamicFiltering(context.getSession())` check seems redundant as planner should not add dynamic filter when dynamic filtering is disabled